integrations-and-supported-tools/detectron2/scripts/Neptune_detectron2.py

```diff
@@ -0,0 +1,147 @@
+    MetadataCatalog.get("balloon_" + d).set(thing_classes=["balloon"])
+balloon_metadata = MetadataCatalog.get("balloon_train")
+
+
+# Configure the model for training
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+cfg = get_cfg()
+cfg.MODEL.DEVICE = device
+cfg.DATALOADER.NUM_WORKERS = 0
+cfg.merge_from_file(
+    model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")
+)
+cfg.DATASETS.TRAIN = ("balloon_train",)
+cfg.DATASETS.TEST = ()
+cfg.DATALOADER.NUM_WORKERS = 2
+cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(
+    "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"
+)  # Let training initialize from model zoo
+cfg.SOLVER.IMS_PER_BATCH = 2  # This is the real "batch size" commonly known to deep learning people
+cfg.SOLVER.BASE_LR = 0.00025  # pick a good LR
+cfg.SOLVER.MAX_ITER = 30  # 30 iterations seems good enough for this toy dataset; you will need to train longer for a practical dataset
+cfg.SOLVER.STEPS = []  # do not decay learning rate
+cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128  # The "RoIHead batch size". 128 is faster, and good enough for this toy dataset (default: 512)
+cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1  # only has one class (ballon). (see https://detectron2.readthedocs.io/tutorials/datasets.html#update-the-config-for-new-datasets)
+# NOTE: this config means the number of classes, but a few popular unofficial tutorials incorrect uses num_classes+1 here.
+
+
+# (Neptune) Create a hook using Neptune integration
+hook = NeptuneHook(run=run, log_checkpoints=True, log_model=True)
+
+
+# (Neptune) Train the model with hook
+os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
+trainer = DefaultTrainer(cfg)
+trainer.resume_or_load(resume=False)
+trainer.register_hooks([hook])
+trainer.train()
+
+
+# Prepare model for prediction
+cfg.MODEL.WEIGHTS = os.path.join(
+    cfg.OUTPUT_DIR, "model_final.pth"
+)  # path to the model we just trained
+cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.7  # set a custom testing threshold
+predictor = DefaultPredictor(cfg)
+
+
+# (Neptune) Log model predictions to neptune
+dataset_dicts = get_balloon_dicts("balloon/val")
+for idx, d in enumerate(random.sample(dataset_dicts, 3)):
+    im = cv2.imread(d["file_name"])
+    outputs = predictor(
+        im
+    )  # format is documented at https://detectron2.readthedocs.io/tutorials/models.html#model-output-format
+    v = Visualizer(
+        im[:, :, ::-1],
+        metadata=balloon_metadata,
+        scale=0.5,
+        instance_mode=ColorMode.IMAGE,  # remove the colors of unsegmented pixels. This option is only available for segmentation models
+    )
+    out = v.draw_instance_predictions(outputs["instances"].to("cpu"))
+    image = out.get_image()[:, :, ::-1]
+    img_rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+    run["training/prediction_visualization"][idx].upload(
+        neptune.types.File.as_image(img_rgb / 255.0)
+    )
```

Comment:
Could we add `run.stop()` at the end here? IIRC the hook was changed to not stop the run explicitly.

We had a discussion with Mycek that it's better if our scripts also include stopping, as it's a best practice to stop objects when they're no longer needed. Helps "teach" this aspect to our users and waterproofs things a bit if they copy code around.

@SiddhantSadangi Do you have some guidelines for example scripts somewhere? I can also cover it in the Code Style Guide (under **Product > Documentation** in Notion).

Reply:
We don't have guidelines for scripts iirc.
The point about adding `stop()` to scripts makes sense. I'll create a task for it